# HG changeset patch
# Parent 5880e4546bd5afe208a4bb2cfb2c72cde1863b8b
# User sbr
# Date 1349604320 -7200

Don't show unloading indicator if order is 'no unloading' and 'no loading'.

diff -r 5880e4546bd5 src/lang/english.txt
--- a/src/lang/english.txt	Fri Oct 05 19:16:23 2012 +0000
+++ b/src/lang/english.txt	Sun Oct 07 18:30:12 2012 +0200
@@ -3757,6 +3757,8 @@
 STR_PERCENT_DOWN                                                :{WHITE}{NUM}%{DOWN_ARROW}
 STR_PERCENT_UP_DOWN_SMALL                                       :{TINY_FONT}{WHITE}{NUM}%{UP_ARROW}{DOWN_ARROW}
 STR_PERCENT_UP_DOWN                                             :{WHITE}{NUM}%{UP_ARROW}{DOWN_ARROW}
+STR_PERCENT_NONE_SMALL                                          :{TINY_FONT}{WHITE}{NUM}%
+STR_PERCENT_NONE                                                :{WHITE}{NUM}%
 
 # Income 'floats'
 STR_INCOME_FLOAT_COST_SMALL                                     :{TINY_FONT}{RED}Cost: {CURRENCY_LONG}
diff -r 5880e4546bd5 src/vehicle.cpp
--- a/src/vehicle.cpp	Fri Oct 05 19:16:23 2012 +0000
+++ b/src/vehicle.cpp	Sun Oct 07 18:30:12 2012 +0200
@@ -1259,6 +1259,8 @@
 	if (colour != NULL) {
 		if (unloading == 0 && loading) {
 			*colour = STR_PERCENT_UP;
+		} else if (unloading == 0 && !loading) {
+			*colour = STR_PERCENT_NONE;
 		} else if (cars == unloading || !loading) {
 			*colour = STR_PERCENT_DOWN;
 		} else {
